### ISSUE TITLE
win_key_down : always try layout

### ISSUE THREAD

**Issue by maxime1986** [mintty/mintty#483]
Hello,
I have a bépo keyboard with a specific layout where I mapped the VK_SCROLL key to the '&' charactere.
This specific key doesn't work in mintty.
After some digging I found that the "layout" function was never called on this key in win_key_down.
win_key_down should always check if the key is handled by the layout.

I propose this patch : 

```diff
diff --git a/src/wininput.c b/src/wininput.c
index 22c5db5..ebd975f 100644
--- a/src/wininput.c
+++ b/src/wininput.c
@@ -772,10 +772,9 @@ win_key_down(WPARAM wp, LPARAM lp)
       else if (key <= '9') app_pad_code(key);
       else if (VK_OEM_PLUS <= key && key <= VK_OEM_PERIOD)
         app_pad_code(key - VK_OEM_PLUS + '+');
-    when VK_PACKET:
-      layout();
     otherwise:
-      return 0;
+      if (!layout())
+        return false;
   }
 
   hide_mouse();
```

**Comment by mintty** [comment]
Thank you. I'm changing that.

**Comment by mintty** [comment]
Released in 2.2.1.
